### ISSUE TITLE
[otel-targetallocator]  Unable to pass scrape_protocols

### ISSUE THREAD

**Issue by halleystar** [open-telemetry/opentelemetry-operator#3035]
### Component(s)

target allocator

## Description
In k8s, I use opentelemetry-operator and servicemonitor to collect metrics. I want to use opentelemetry's OTEL exponential histograms. But there is a bug in it that makes my purpose impossible to achieve.
## Steps to Reproduce
My config is like this:
![image](https://github.com/open-telemetry/opentelemetry-operator/assets/5334156/6ae4e6e8-ae5c-4285-b0c3-9dfa0a9a8f3a)

## Expected Result 
export OTEL exponential histograms.
## Actual Result
export classic histograms.

### Kubernetes Version

1.28.0

### Operator version

v0.102.0

### Collector version

v0.102.0

## Environment
OS: (e.g., "Ubuntu 20.04")
Compiler(if manually compiled): (e.g., "go 14.2")

**Comment by swiatekm-sumo** [comment]
Could you go into more detail as to what bug you're experiencing, and why you believe you need to set scrape_protocols in the receiver configuration? They can be set on the ServiceMonitor, and this should work correctly. Is the issue that they can't be set globally?

**Comment by halleystar** [comment]
I want to add:
``` 
scrapeProtocols:
  - PrometheusProto
  - OpenMetricsText1.0.0
  - OpenMetricsText0.0.1
  - PrometheusText0.0.4
```


to use opentelemetry's OTEL exponential。Regardless of whether it is placed in servicemonitor in opentelemetry operator. when i put it in servicemonitor, it also invalid:
![image](https://github.com/open-telemetry/opentelemetry-operator/assets/5334156/d6d6d24c-0b8a-4e09-a01d-a92ece7c14b2)

**Comment by swiatekm-sumo** [comment]
Thanks for the demonstration. I investigated, and the reason this doesn't work is that our version of prometheus-operator doesn't support setting this field yet. Support was added in [prometheus-operator 0.72.0](https://github.com/prometheus-operator/prometheus-operator/blob/v0.72.0/CHANGELOG.md#0720--2024-02-23), and we're on `0.71.2`.

The reason we can't upgrade yet is that this requires Go 1.22, which we can't require until Go 1.21 becomes unsupported, as per #2757. @jaronoff97 @pavolloffay I believe this is the first time this policy has blocked a user requested feature.
